quarkus.ls/com.redhat.quarkus.ls/src/main/java/com/redhat/quarkus/services/QuarkusCompletions.java

```diff
@@ -204,8 +207,77 @@ private static String getPropertyName(String propertyName, boolean snippetsSuppo
 	 * @param completionSettings the completion settings
 	 * @param list               the completion list to fill
 	 */
-	private static void collectPropertyValueSuggestions(Node node, QuarkusProjectInfo projectInfo,
-			QuarkusCompletionSettings completionSettings, CompletionList list) {
+	private static void collectPropertyValueSuggestions(Node node, PropertiesModel model,
+			QuarkusProjectInfo projectInfo, QuarkusCompletionSettings completionSettings, CompletionList list) {
+
+		Node parent = node.getParent();
+		if (parent != null && parent.getNodeType() != Node.NodeType.PROPERTY) {
+			return;
+		}
+
+		PropertyKey key = getPropertyKeyFromProperty((Property) parent);
+
+		if (key == null) {
+			return;
+		}
+
+		String keyText = key.getText().trim();
+
+		ExtendedConfigDescriptionBuildItem property = 
+			projectInfo.getProperty(keyText);
+
+		Collection<String> enums = getEnums(property);
+
+		if (enums != null && !enums.isEmpty()) {
+			for (String e: enums) {
+				list.getItems().add(getValueCompletionItem(e, node, model));
+			}
+		}
+	}
+
+	/**
+	 * Returns the <code>PropertyKey</code> object associated with 
+	 * <code>property</code>.
+	 * 
+	 * @param property the property to find the <code>PropertyKey</code> for
+	 * @return         the <code>PropertyKey</code> object associated with 
+	 * <code>property</code>
+	 */
+	private static PropertyKey getPropertyKeyFromProperty(Property property) {
+		Node key = property.getKey();
+
+		if (key == null) {
+			return null;
+		}
+
+		return (PropertyKey) key;
+	}
+
+	/**
+	 * Returns the <code>CompletionItem</code> which offers completion for value
+	 * completion for <code>value</code> at the start offset of <code>node</code>.
+	 * 
+	 * @param value the value for completion
+	 * @param node  the node where its start offset is where value completion occurs
+	 * @param model the property model
+	 * @return      the value completion item
+	 */
+	private static CompletionItem getValueCompletionItem(String value, Node node, PropertiesModel model) {
+		CompletionItem completionItem = new CompletionItem(value);
+		completionItem.setKind(CompletionItemKind.Value);
+
+		Range range = null;
+		try {
+			TextDocument doc = model.getDocument();
+			int startOffset = node.getStart();
+			range = doc.lineRangeAt(startOffset);

```

Comment:
Please don't use doc to retrieve range. Please use property model. I mean Property parent getEnd

Reply:
Ok forget my comments